here-naksha-lib-handlers/src/test/java/com/here/naksha/lib/handlers/TagFilterHandlerTest.java

```diff
@@ -56,71 +62,75 @@ void testReadWithoutPOpAndMultipleTagFilters() {
         // When: a function is called with request and tag filter configuration
         TagFilterHandler.applyFilterConditionOnRequest(request, tagFilter);
         // Then: Validate that Tags "and" condition is added to request with multiple tag filters
-        POpAssertion.assertThatOperation(request.getPropertyOp())
-                .hasType(POpType.AND)
-                .hasChildrenThat(
-                        op1 -> op1.hasType(POpType.EXISTS).hasPRefWithPath(TAGS_PROP_PATH).existsWithTagName("violated_ftype_topology"),
-                        op2 -> op2.hasType(POpType.EXISTS).hasPRefWithPath(TAGS_PROP_PATH).existsWithTagName("some_other_tag")
-                )
-        ;
+        final ITagQuery tagQuery = request.getQuery().getTags();
+        Assertions.assertInstanceOf(TagAnd.class, tagQuery);
+        List<String> tagNameList = ((TagAnd) tagQuery).stream()
+                .map(tagOp -> (TagExists) tagOp)
+                .map(TagExists::getName).toList();
+        assertEquals(tagFilter.size(), tagNameList.size());
+        Assertions.assertTrue(tagNameList.contains("violated_ftype_topology"));
+        Assertions.assertTrue(tagNameList.contains("some_other_tag"));
     }
 
     @Test
+        //TODO might be redundant now that property query and tag query are separate
```

Comment:
TODOs should be actionable - we either remove the TODO or the test.
I would remove the TODO - it might be redundant but it's already written - one more test won't harm I think 🤷